crates/sui-data-ingestion-core/src/lib.rs

```diff
@@ -26,7 +28,18 @@ pub use worker_pool::WorkerPool;
 #[async_trait]
 pub trait Worker: Send + Sync {
     type Result: Send + Sync + Clone;
-    async fn process_checkpoint(&self, checkpoint: &CheckpointData) -> Result<Self::Result>;
+    async fn process_checkpoint_arc(
+        &self,
+        checkpoint: Arc<CheckpointData>,
```

Comment:
One maybe minor nit here would be to have this take a `&Arc<CheckpointData>` so that you don't need to clone the arc unless the implementor needs to do so